pkg/registry/apis/provisioning/resources/parser.go

```diff
@@ -167,6 +167,12 @@ func (r *Parser) Parse(ctx context.Context, info *repository.FileInfo, validate
 		Checksum: info.Hash,
 	})
 
+	if obj.GetName() == "" {
```

Comment:
don't you think we should still allow requests to POST /files without that type of metadata? same as we don't force anybody to pass the namespace or other annotations and how k8s can generate IDs for you.

Reply:
👀  maybe we can use the dryRun result when `generateName` is set...